pkg/drain/nodegroup.go

```diff
@@ -144,18 +145,31 @@ func NodeGroup(clientSet kubernetes.Interface, ng *api.NodeGroup, waitTimeout ti
 
 			for _, node := range nodes.Items {
 				if newPendingNodes.Has(node.Name) {
+					retry := 1
 					pending, err := evictPods(drainer, &node)
-					if err != nil {
-						return err
-					}
-					logger.Debug("%d pods to be evicted from %s", pending, node.Name)
-					if pending == 0 {
-						drainedNodes.Insert(node.Name)
+					for pending > 0 {
+						logger.Debug("%d pods to be evicted from %s", pending, node.Name)
+						time.Sleep(5 * time.Second)
+						pending, err = evictPods(drainer, &node)
+						if err != nil && retry < 60 {
+							logger.Warning("pod eviction error: \"%s\", on node: %s (retry in 5 sec, %d/60)", err, node.Name, retry)
+						} else if retry >= 60 {
+							logger.Warning("pod eviction unable to complete after 60 retry on node: %s", node.Name)
```

Comment:
```suggestion
							logger.Warning("pod eviction unable to complete after 60 retries on node: %s", node.Name)
```

Reply:
fixed

Reply:
If something is missing, let me know.
Thanks